src/Contracts/CoreApi/Endpoint/Data/DataInterface.php

```diff
@@ -48,4 +53,18 @@ public function replace(string $ouuid, array $rawData): DraftInterface;
      * @throws CoreApiExceptionInterface
      */
     public function update(string $ouuid, array $rawData): DraftInterface;
+
+    /**
+     * @param array<string, mixed> $rawData
+     *
+     * @throws CoreApiExceptionInterface
+     */
+    public function createOrUpdateAndFinalize(string $ouuid, array $rawData): void;
+
+    /**
+     * @param array<string, mixed> $rawData
+     *
+     * @throws CoreApiExceptionInterface
+     */
+    public function createOrReplaceAndFinalize(string $ouuid, array $rawData): void;
```

Comment:
Do not like these names, we need to discuss.

- Update or replace we could do with a third parameter and put it default on update. ?
- They also return void, I would be expecting the RevisionInterface.